subprojects/core/src/main/java/org/gradle/api/internal/file/UnionFileTree.java

```diff
@@ -52,13 +52,11 @@ public void visitContents(FileCollectionResolveContext context) {
         context.add(sourceTrees);
     }
 
-    @Override
-    public UnionFileTree add(FileCollection source) {
+    public void uniteWith(FileCollection source) {
```

Comment:
I don't really like `uniteWith`, since we are not actually "uniting" the `UnionFile{Tree,Collection}` with the file collection. We add another file collection to the union.

How about calling the method `addToUnion`? `addSourceCollection` could also be another possiblity.
WDYT?

Reply:
Changed it to `addToUnion()`.